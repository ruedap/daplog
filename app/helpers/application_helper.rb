module ApplicationHelper
  def blog_title(article_title = nil)
    if article_title.present?
      "#{article_title} - #{blog_name}"
    else
      blog_name
    end
  end

  def blog_description(description = nil)
    if description
      truncate(strip_tags(description).strip.gsub(/\n/, ''), length: 130)
    else
      'できればググってもでてこないようなことだけをかきたいけれど'
    end
  end

  def blog_author
    'ruedap'
  end

  def blog_ogp_image
    image_url("ogp-#{@hue}.png")
  end

  private

  def blog_name
    'アインシュタインの電話番号'
  end
end
