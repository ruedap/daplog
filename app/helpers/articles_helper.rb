module ArticlesHelper
  def generate_article_link_list(article_data_array)
    content_tag :ul do
      article_year = ''
      article_data_array.each do |d|
        path = d[:path].sub(/.+?(\d{4})-(\d{2})-(\d{2})-(.+?)\.md/, "\/\\1\/\\2\/\\3\/\\4\/")
        year = d[:date][0..3]
        html = ''
        unless article_year == year
          html << '</ul>' unless article_year == ''
          article_year = year
          html << content_tag(:li, article_year)
          html << '<ul>'
        end
        html << content_tag(:li) do
          content_tag(:time, d[:date], datetime: d[:time]) +
          link_to(d[:title], path)
        end
        concat html.html_safe
      end
    end
  end
end
