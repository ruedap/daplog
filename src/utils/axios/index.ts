import axios from 'axios'

export const axiosJson = axios.create({
  baseURL: '/api',
  headers: {
    accept: 'application/json'
  }
})
