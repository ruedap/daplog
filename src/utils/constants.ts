// TODO: i18n
export const BLOG_NAME = 'アインシュタインの電話番号'
export const PORTFOLIO_SITE_NAME = 'ルエダップコム'
