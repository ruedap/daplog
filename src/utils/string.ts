import { TArticlePath } from '@src/types'

export const stripHtmlTags = (htmlStr: string): string => {
  return htmlStr.replace(/(<([^>]+)>)/ig, '')
}

const removeExt = (str: string, ext: string = '.md'): string => {
  return str.replace(ext, '')
}

const id2Url = (id: string): string => {
  return id.replace(/(\d{4})-(\d{2})-(\d{2})-(.*)/, '$1/$2/$3/$4')
}

export const id2Prams = (id: string): TArticlePath => {
  const a = id2Url(id).split('/')
  return { year: a[0], month: a[1], date: a[2], title: a[3]} as TArticlePath
}

export const normalizeArticleUrl = (path: string, ext: string = '.md'): string => {
  return id2Url(removeExt(path, ext))
}

export const id2DateString = (id: string):string => {
  return id.slice(0, 10)
}
