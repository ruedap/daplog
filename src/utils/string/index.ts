import he from 'he'

export const stripHtmlTags = (htmlStr: string): string => {
  const d = document.createElement('div')
  d.innerHTML = he.unescape(htmlStr)
  const result = String(d.textContent || d.innerText)

  return result
}

const removeExt = (str: string, ext: string): string => {
  // return str.split('.').slice(0, -1).join('.')
  return str.replace(ext, '')
}

const dateHyphen2slash = (str: string): string => {
  return str.replace(/(\d{4})-(\d{2})-(\d{2})-(.*)/, '$1/$2/$3/$4')
}

export const normalizeArticlePath = (base: string, ext: string): string => {
  return dateHyphen2slash(removeExt(base, ext))
}
