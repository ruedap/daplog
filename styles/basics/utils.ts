import { createGlobalStyle, css } from 'styled-components'

const prefix = 'u'

export const StyledBasicUtils = createGlobalStyle`
  ${({ theme }) => css`
    ${
      Object.entries({ margin: 'm', padding: 'p' }).map(([prop, abbr]) => {
        return Object.entries(theme.space).map(([size, unitize]) => {
          return `
            .${prefix}-${abbr}-${size} { ${prop}: ${unitize.px()} !important; }
            .${prefix}-${abbr}t-${size},
            .${prefix}-${abbr}y-${size} {
              ${prop}-top: ${unitize.px()} !important;
            }
            .${prefix}-${abbr}r-${size},
            .${prefix}-${abbr}x-${size} {
              ${prop}-right: ${unitize.px()} !important;
            }
            .${prefix}-${abbr}b-${size},
            .${prefix}-${abbr}y-${size} {
              ${prop}-bottom: ${unitize.px()} !important;
            }
            .${prefix}-${abbr}l-${size},
            .${prefix}-${abbr}x-${size} {
              ${prop}-left: ${unitize.px()} !important;
            }
          `
        })
      })
    }
  `}
`
