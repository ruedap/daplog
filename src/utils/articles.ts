import fs from 'fs'
import path from 'path'
import remark from 'remark'
import html from 'remark-html'
import { stripHtmlTags, getTitleHtml, id2DateString } from '@src/utils/string'
import { TArticleItem } from '@src/types'

const PATH = 'src/articles'
const articlesDirectory = path.join(process.cwd(), PATH)

export const getSortedArticleList = () => {
  const fileNames = fs.readdirSync(articlesDirectory)
  const allArticlesData = fileNames.map(fileName => {
    const id = fileName.replace(/\.md$/, '')
    const fullPath = path.join(articlesDirectory, fileName)
    const fileContents = fs.readFileSync(fullPath, 'utf8')
    const contentHtml = markdown2Html(fileContents)
    const title = stripHtmlTags(getTitleHtml(contentHtml))
    const date = id2DateString(id)

    return {
      id,
      date,
      title
    } as TArticleItem
  })
  
  // TODO: sort
  return allArticlesData.reverse()
}

export function getAllArticleIds() {
  const fileNames = fs.readdirSync(articlesDirectory)

  return fileNames.map(fileName => {
    return {
      params: {
        id: fileName.replace(/\.md$/, '')
      }
    }
  })
}

export function getArticleData(id: string) {
  const fullPath = path.join(articlesDirectory, `${id}.md`)
  const fileContents = fs.readFileSync(fullPath, 'utf8')
  const contentHtml = markdown2Html(fileContents)
  const title = getTitleHtml(contentHtml)
  const date = id2DateString(id)

  return {
    id,
    title,
    date,
    contentHtml,
  }
}

const markdown2Html = (markdownString: string): string => {
  const processedContent = remark()
    .use(html)
    .processSync(markdownString)
  const contentHtml = processedContent.toString()
  
  return contentHtml
}
