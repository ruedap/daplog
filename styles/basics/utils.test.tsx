import { StyledBasicUtils } from './utils'
import 'jest-styled-components'
import renderer from 'react-test-renderer'
import { renderWithTheme } from '@/tests/helpers'

describe('StyledBasicUtils', () => {
  it('should be matched snapshot', () => {
    const actual = renderWithTheme(<StyledBasicUtils />).toJSON()
    expect(actual).toMatchSnapshot()
  })
})
