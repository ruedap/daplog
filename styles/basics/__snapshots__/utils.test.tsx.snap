// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledBasicUtils should be matched snapshot 1`] = `null`;
